consensus/parlia/snapshot.go

```diff
@@ -315,20 +367,16 @@ func (s *Snapshot) apply(headers []*types.Header, chain consensus.ChainHeaderRea
 					}
 				}
 			}
-			oldLimit := len(snap.Validators)
-			newLimit := len(newVals)
-			if newLimit < oldLimit {
-				for i := 0; i < oldLimit-newLimit; i++ {
-					delete(snap.RecentForkHashes, number-uint64(newLimit)-uint64(i))
-				}
-			}
 			snap.Validators = newVals
 			if chainConfig.IsLuban(header.Number) {
 				validators := snap.validators()
 				for idx, val := range validators {
 					snap.Validators[val].Index = idx + 1 // offset by 1
 				}
 			}
+			for i := snap.versionHistoryCheckLen(); i < oldVersionsLen; i++ {
```

Comment:
this `apply(...)` is not code review friendly...

Reply:
we can compare the logic about `deleting RecentForkHashes`,
the effect is the same as before, but the code is much more simplified